### ISSUE TITLE
Support for sticky items

### ISSUE THREAD

**Issue by petyosi** [petyosi/react-virtuoso#2]
Marking certain indices as sticky can be used to create groups. The rendering itself won't change, but the implementation should switch the pinned items to the closest sticky one.

**Comment by oisinlavery** [comment]
Just to confirm does this mean you are emulating css `position: sticky` behavior?

I don't think any of the other virtualization libraries do this so it would be a great feature to support.

**Comment by petyosi** [comment]
That's my intent, yes. In a nutshell, the component should behave like the iOS listview.

**Comment by petyosi** [comment]
Just published `v0.3.0` that includes a `GroupedVirtuoso` component that supports sticky group headers. Check the [release notes](https://github.com/petyosi/react-virtuoso/releases/tag/v0.3.0).

To clarify something - my initial idea was to export a "lower" level API that accepts sticky items indices and leave the grouping logic to the client. However, after building one example with that, I realized how much of a pain that would be - one has to manually merge the "regular" items with groups and to include complex rendering logic in the `item` prop. 

So, I decided to make the grouping convenient and explicit, assuming that that's what most people would try to do anyway. 

Feedback is welcome. [Check the live demo](https://virtuoso.dev/?path=/story/grouping--grouped-by-first-letter). Cheers!
